rescript/parse_rdp.py

```diff
@@ -0,0 +1,89 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2021, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import pandas as pd
+from collections import OrderedDict, defaultdict
+from q2_types.feature_data import DNAIterator
+from .parse_silva_taxonomy import _keep_allowed_chars
+
+
+ALLOWED_RDP_RANKS = OrderedDict({'rootrank': 'root__', 'domain': 'd__',
+                                 'phylum': 'p__', 'class': 'c__',
+                                 'subclass': 'cs__', 'order': 'o__',
+                                 'suborder': 'os__', 'family': 'f__',
+                                 'genus': 'g__'})
+
+DEFAULT_RDP_RANKS = ['domain', 'phylum', 'class', 'subclass', 'order',
+                     'suborder', 'family', 'genus']
+
+
+def parse_rdp_taxonomy(rdp_reference_sequences: DNAIterator,
+                       rank_propagation: bool = True, ranks: list = None,
+                       include_species_labels: bool = False) -> pd.Series:
+
+    sp_label_dict = defaultdict()
+    taxd = defaultdict()
+    for seq in rdp_reference_sequences:
+        # parse taxonomy
+        seq_id = seq.metadata['id']
+        sp_label, tax_data = seq.metadata['description'].split(
+                                'Lineage=')
+        new_sp_label = _keep_allowed_chars(
+                                        "_".join(sp_label.strip().split()[:2]))
+        sp_label_dict[seq_id] = new_sp_label
```

Comment:
it's a little confusing that species labels are kept in a separate dict from the rest of the taxonomy labels. Why not just parse and add the species labels directly in the taxonomy dict (if include_species_labels is True)?

wouldn't this also be needed for proper rank_propagation? (i.e., in any cases where the species ID is missing?)